packages/swapper/src/swappers/thorchain/buildThorTrade/buildThorTrade.ts

```diff
@@ -70,14 +86,17 @@ export const buildTrade = async ({ deps, input }: BuildTradeArgs): Promise<ThorT
         buyAssetTradeFeeUsd: quote.feeData.buyAssetTradeFeeUsd,
       })
 
-      return {
+      if (maybeEthTradeTx.isErr()) return Err(maybeEthTradeTx.unwrapErr())
```

Comment:
Ditto, `match`.

Reply:
See this comment and conversation: 917175dee5a35b68e4345dbddd71a4f983c43b27

Updated all `unwrapErr()` / `unwrap()` instances that aren't async to `.map()` in 917175dee5a35b68e4345dbddd71a4f983c43b27.
Let's do another pass in a follow-up and iterate on how we can achieve the same for async flows? That's not going to work in such a straightforward with async flows way due to the way promises are boxed into a `Result`, unfortunately.